includes/data-stores/class-wc-customer-download-data-store.php

```diff
@@ -68,38 +72,46 @@ public function create( &$download ) {
 	/**
 	 * Method to read a download permission from the database.
 	 *
-	 * @param $download
+	 * @param WC_Customer_Download $download WC_Customer_Download object.
 	 *
-	 * @throws Exception
+	 * @throws Exception Throw exception if invalid download is passed.
 	 */
 	public function read( &$download ) {
 		global $wpdb;
 
 		$download->set_defaults();
+		$raw_download = $wpdb->get_row(
```

Comment:
Probably better check if ID is valid before trying to do some query.

Reply:
@claudiosanches, on commit 56cb3d2d4f358488a7a7240175a49ecd26bab92b I moved the `! $download->get_id()` check to happen before the query is executed. What I like about this approach is that we don't run a query that is not needed. What I don't like is that I'm duplicating the exception, but I don't see any way around it. Please let me know if you have a suggestion or if it is good as it is now. Thanks.